upf/upf_pfcp_api.c

```diff
@@ -2505,7 +2551,7 @@ handle_session_modification_request (pfcp_msg_t * msg,
   if (r == 0)
     resp->cause = PFCP_CAUSE_REQUEST_ACCEPTED;
 
-  upf_pfcp_send_response (msg, &resp_dmsg);
+  upf_pfcp_send_response (msg, &resp_dmsg, outlives_session);
```

Comment:
there is on thing that bugs me about the `outlives_session` argument, why do we keep a reference between session and messages in the response cache and request queue anyway?
There should be no reason to remove a message from the response cache even when the session is deleted. A possible resent of the response does not need to access the session in any way.
The request queue is a bit more complex,
any answer should remove the message from the queue, but it should not attempt to access the session if the session is already gone.

Reply:
Removed the code that was dropping session-related messages upon session deletion, and updated the tests accordingly.